src/karapace/schema_reader.py

```diff
@@ -433,9 +443,8 @@ def consume_messages(self, msgs: list[Message], watch_offsets: bool) -> None:
             # Default keymode is CANONICAL and preferred unless any data consumed
             # has key in non-canonical format. If keymode is set to DEPRECATED_KARAPACE
             # the subsequent keys are omitted from detection.
-            if not self.ready and self.key_formatter.get_keymode() == KeyMode.CANONICAL:
-                if msg_keymode == KeyMode.DEPRECATED_KARAPACE:
-                    self.key_formatter.set_keymode(KeyMode.DEPRECATED_KARAPACE)
+            if self.key_formatter.get_keymode() == KeyMode.CANONICAL and msg_keymode == KeyMode.DEPRECATED_KARAPACE:
+                self.key_formatter.set_keymode(KeyMode.DEPRECATED_KARAPACE)
```

Comment:
I think its better to keep the behaviour in that way, otherwise based on the restarts the cluster can behave in different ways. Lets make an example:
1. Karapace starts, all the messages are `CANONICAL`
2. Karapace become ready
3. Someone produce a `DEPRECATED` record (its an illegal behaviour but its possible to have it in production). Now, since `ready` its `True` Karapace will produce always only `CANONICAL` records
4. Karapace restarts, repeat the read
5. Now it produces only `DEPRECATED` records


We have proved that this "state" its not based on the input but on the sequence of events and its not stable, a restart can change the behaviour of the node causing an extremely hard issue to detect a runtime.

The proper fix should be to keep the info about witch key mode to use in the message itself, so karapace will always select the right formatter once a record its updated/changed/deleted